Build failure: v2ray-geoip

### Steps To Reproduce

Steps to reproduce the behavior:
1. On da443e5366f1ccf5dd940f3d145a4b7835091a20, build *v2ray-geoip*

Probably we can pin to buildGo120Module?

### Build log

```
this derivation will be built:
  /nix/store/9j0xf506iw1pv5fil4lah2rxp65dlbpz-v2ray-geoip-2023-10.drv
building '/nix/store/9j0xf506iw1pv5fil4lah2rxp65dlbpz-v2ray-geoip-2023-10.drv'...
unpacking sources
unpacking source archive /nix/store/6hflxbzxyydiqzn290bxl1vzb06cjsph-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
panic: qtls.ClientSessionState doesn't match

goroutine 1 [running]:
github.com/quic-go/qtls-go1-20.init.0()
        github.com/quic-go/qtls-go1-20@v0.1.1/unsafe.go:14 +0x2d8
error: builder for '/nix/store/9j0xf506iw1pv5fil4lah2rxp65dlbpz-v2ray-geoip-2023-10.drv' failed with exit code 2;
```

### Notify maintainers

@NickCao

### Metadata

https://github.com/NixOS/nixpkgs/commit/da443e5366f1ccf5dd940f3d145a4b7835091a20

--- a/pkgs/data/misc/v2ray-geoip/default.nix
+++ b/pkgs/data/misc/v2ray-geoip/default.nix
@@ -10,16 +10,16 @@
 let
   generator = pkgsBuildBuild.buildGoModule {
     pname = "v2ray-geoip";
-    version = "unstable-2023-03-27";
+    version = "unstable-2023-10-11";
 
     src = fetchFromGitHub {
       owner = "v2fly";
       repo = "geoip";
-      rev = "9321a7f5e301a957228eba44845144b4555b6658";
-      hash = "sha256-S30XEgzA9Vrq7I7REfO/WN/PKpcjcI7KZnrL4uw/Chs=";
+      rev = "3182dda7b38c900f28505b91a44b09ec486e6f36";
+      hash = "sha256-KSRgof78jScwnUeMtryj34J0mBsM/x9hFE4H9WtZUuM=";
     };
 
-    vendorHash = "sha256-bAXeA1pDIUuEvzTLydUIX6S6fm6j7CUQmBG+7xvxUcc=";
+    vendorHash = "sha256-rlRazevKnWy/Ig143s8TZgV3JlQMlHID9rnncLYhQDc=";
 
     meta = with lib; {
       description = "GeoIP for V2Ray";

--- a/pkgs/data/misc/v2ray-geoip/default.nix
+++ b/pkgs/data/misc/v2ray-geoip/default.nix
@@ -8,7 +8,7 @@
 }:
 
 let
-  generator = pkgsBuildBuild.buildGoModule {
+  generator = pkgsBuildBuild.buildGo120Module {
     pname = "v2ray-geoip";
     version = "unstable-2023-10-11";
 

